bin/utils.dart

```diff
@@ -0,0 +1,33 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:io/io.dart';
+
+class PresubmitUtils {
+  ProcessManager manager;
+
+  PresubmitUtils() {
+    this.manager = new ProcessManager();
+  }
+
+  /// Fetches the configlet file if it doesn't exist already.
+  Future<Null> fetchConfiglet() async {
+    File configletFile = new File('bin/configlet');
+
+    if (!configletFile.existsSync()) {
+      print('Fetching configlet...');
+      Process spawn = await manager.spawn('bin/fetch-configlet', []);
+      await spawn.exitCode;
+    }
+  }
+
+  /// Runs the [executable] with [arguments].
+  Future<Null> run(String executable, [List<String> arguments = const []]) async {
+    Process spawn = await manager.spawn(executable, arguments);
```

Comment:
Before you spawn the `executable` check whether or not it is in fact executable using `isExecutable`, If it's not  
```dart
print("The $executable is not executable, verify you have sufficient permissions to execute it.");
```
and `return false`.
You can checkout [this ](https://github.com/matanlurey/dart-future-examples/blob/master/bin/5_using_future_or.dart)example of `FutureOr` for this purpose.

Reply:
Windows doesn't have a concept of executables, but for other OSs we need to check this.

Reply:
I added the `isExecutable` check before running the command, but for some reason is always returns `false` when running `pub` commands. Probably because the user that runs the dart script on my system doesn't have permission to run `pub`, I don't know.